### PR TITLE
uniq arrays model mop up (SYN-1393, SYN-2251)

### PR DESCRIPTION
**Pull request by whippit**

### CODE REVIEW

File: synapse/lib/modelrev.py [hunk 1]
```diff
@@ -64,6 +65,54 @@ async def save():
...
+            'it:mitre:attack:technique:tactics',
+            'it:mitre:attack:software:techniques',
+            'it:mitre:attack:mitigation:addresses',
+
+            'it:reveng:function:impcalls',
+            'it:reveng:filefunc:funccalls',
+
+            'risk:vuln:cwes',
+
+            'syn:cmd:input',
```
Comment (vEpiphyte) [inline synapse/lib/modelrev.py:108]: These are runt nodes

**Review by vEpiphyte — DISMISSED** [review]

File: synapse/lib/modelrev.py [hunk 2]
```diff
@@ -64,6 +65,43 @@ async def save():
...
+            'crypto:x509:cert:ext:sans',
+            'crypto:x509:cert:ext:crls',
+            'crypto:x509:cert:identities:fqdns',
+            'crypto:x509:cert:identities:emails',
+            'crypto:x509:cert:identities:ipv4s',
+            'crypto:x509:cert:identities:ipv6s',
+            'crypto:x509:cert:identities:urls',
+            'crypto:x509:cert:crl:urls',
+
+            'inet:email:message:headers',
```
Comment (invisig0th) [inline synapse/lib/modelrev.py:82]: Since they can be duplicate within the protocol should we maintain their order and their ability to be duplicate here?

**Review by invisig0th — CHANGES_REQUESTED** [review]

File: synapse/lib/modelrev.py (hunk 2 above)
Comment (whippit) [reply synapse/lib/modelrev.py]: Removed in latest push.

**Review by invisig0th — APPROVED** [review]

File: synapse/models/infotech.py [hunk 3]
```diff
@@ -693,7 +693,7 @@ def getModelDefs(self):
                     ('url', ('inet:url', {}), {
                         'doc': 'A URL linking this CVE to a full description.',
                     }),
-                    ('references', ('array', {'type': 'inet:url', 'uniq': True}), {
+                    ('references', ('array', {'type': 'inet:url', 'uniq': True, 'sorted': True}), {
```
Comment (vEpiphyte) [inline synapse/models/infotech.py:696]: This is not migrated (it is already uniqued) but not sorted in storage.

Comment (invisig0th) [reply synapse/models/infotech.py:696]: Yep, we should probably migrate these to be ordered in storage

**Review by invisig0th — CHANGES_REQUESTED** [review]

**Review by vEpiphyte — APPROVED** [review]

**Pull request merged**